graphx/src/main/scala/org/apache/spark/graphx/lib/PageRank.scala

```diff
@@ -103,8 +132,14 @@ object PageRank extends Logging {
       // that didn't receive a message. Requires a shuffle for broadcasting updated ranks to the
       // edge partitions.
       prevRankGraph = rankGraph
+      val rPrb = if (personalized) {
+        (src: VertexId ,id: VertexId) => resetProb * delta(src,id)
+      } else {
+        (src: VertexId, id: VertexId) => resetProb
+      }
+
       rankGraph = rankGraph.joinVertices(rankUpdates) {
-        (id, oldRank, msgSum) => resetProb + (1.0 - resetProb) * msgSum
+        (id, oldRank, msgSum) => rPrb(src,id) + (1.0 - resetProb) * msgSum
```

Comment:
This all looks correct but I have a minor concern that the extra function call and branching might increase overhead if the hotspot optimizations don't inline.  Do we have a sense as to the performance cost of this change?   An alternative, less elegant solution would be to have two code paths for lines 141 and 142 depending on whether personalization is enabled.